libraries/Microsoft.Bot.Connector/Authentication/AdalAuthenticator.cs

```diff
@@ -214,9 +220,12 @@ private bool IsAdalServiceUnavailable(Exception ex)
 
             // When the Service Token Server (STS) is too busy because of “too many requests”,
             // it returns an HTTP error 429
-            return adalServiceException.ErrorCode == MsalTemporarilyUnavailable || adalServiceException.StatusCode == 429;
+            return adalServiceException.ErrorCode == MsalTemporarilyUnavailable || adalServiceException.StatusCode == HttpTooManyRequests;
         }
 
+        private bool IsAdalServiceInvalidRequest(Exception ex)
+            => ex is AdalServiceException adal && adal.StatusCode == (int)HttpStatusCode.BadRequest;
```

Comment:
I'm not at all clear on what this line of code is doing. Can you add a nice multi-line comment. 

If nothing else, BadRequest and InvalidRequest are two very different things. 

Also, the complexity of this if statement is staggering, as I've got to know the operator presicdence for:
1. is
2. &&
3. ==
4. (int)

... and understand the more modern C# declare inline syntax. 

Please turn this into a simpler method.